src/internal/m365/onedrive/export.go

```diff
@@ -0,0 +1,169 @@
+package onedrive
+
+import (
+	"context"
+	"errors"
+	"strings"
+
+	"github.com/alcionai/clues"
+
+	"github.com/alcionai/corso/src/internal/data"
+	"github.com/alcionai/corso/src/internal/m365/onedrive/metadata"
+	"github.com/alcionai/corso/src/internal/version"
+	"github.com/alcionai/corso/src/pkg/backup/details"
+	"github.com/alcionai/corso/src/pkg/control"
+	"github.com/alcionai/corso/src/pkg/export"
+	"github.com/alcionai/corso/src/pkg/fault"
+	"github.com/alcionai/corso/src/pkg/path"
+)
+
+var _ export.Collection = &exportCollection{}
+
+// exportCollection is the implementation of export.ExportCollection for OneDrive
+type exportCollection struct {
+	// baseDir contains the path of the collection
+	baseDir string
+
+	// backingCollections will, in most cases contain just one
+	// collection. However, in cases where we have to combine multiple
+	// collections, like when generating pst files for outlook, this
+	// will contain multiple collections.
+	backingCollections []data.RestoreCollection
+
+	// version is the version of the backup this collection was part
+	// of. This is required to figure out how to get the name of the
+	// item.
+	version int
+}
+
+func (ec exportCollection) GetBasePath() string {
+	return ec.baseDir
+}
+
+func (ec exportCollection) GetItems(ctx context.Context) <-chan export.Item {
+	ch := make(chan export.Item)
+
+	go func() {
```

Comment:
This pyramid gets pretty deep.  While it may not seem like much, could you split out this inner func to it's own function?  Then you can call `go items(ctx...)` here.  That gives us two benefits: 1, unnests the func a little; 2, gives us a non-async func that we can call directly for unit and integration testing.